guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,581 @@
+```
+
+Also, now is a good time to remove `ember-welcome-page` from your `package.json`.
+
+Let's put some content in our templates. First, our `application.hbs` will give a friendly greeting and some useful
+information so we know things are working as we proceed.
+
+```handlebars
+<!-- app/templates/application.hbs -->
+
+<h1>Welcome to the Demo App!</h1>
+
+API Key: {{config.apiKey}} <br />
+
+{{#if session.isAuthenticated}}
+    Authenticated
+{{else}}
+    Unauthenticated
+{{/if}}
+
+<hr />
+
+{{outlet}}
+```
+
+The `session` service is automatically injected, so there's no code for that. We'll add the `config` entry for now to
+verify that our `dotenv` configuration is working.
+
+```js
+// app/controllers/application.js
+
+import Ember from 'ember';
+import config from '../config/environment';
+
+export default Ember.Controller.extend({
+  config: config.torii.providers['github-oauth2']
+});
+```
+
+For the `index.hbs` let's put a placeholder until we're ready to fill in real data.
+
+```handlebars
+{{!-- app/templates/index.hbs}}
+
+<h1>Me</h1>
+```
+
+Finally, we'll put a placeholder in `login.hbs`. We'll give it behavior shortly.
+
+```handlebars
+{{!-- app/templates/login.hbs}}
+
+<button>Log in to GitHub</button>
+```
+
+We can ignore the rest of the generated files for this step.
+ 
+You can test the app so far by running `ember serve` and pointing your browser to `http://localhost:4200/` and
+`http://localhost:4200/login`.
+
+### Adding Authentication
+
+We added the configuration for our authentication, but we didn't yet add the pieces that use it. Let's do that first.
+
+```
+ember g authenticator torii
+```
+
+and we'll modify our application and routes with some useful mixins.
+
+First, we'll add the [ApplicationRouteMixin](http://ember-simple-auth.com/api/classes/ApplicationRouteMixin.html) to
+our application route. This is optional, but adds methods supporting the authentication lifecycle that we would
+otherwise have to implement explicitly.
+
+```js
+import Ember from 'ember';
+import ApplicationRouteMixin from 'ember-simple-auth/mixins/application-route-mixin';
+
+export default Ember.Route.extend(ApplicationRouteMixin, {
+});
+```
+
+Next, we'll designate the `index` route as an authenticated route using the
+[AuthenticatedRouteMixin](http://ember-simple-auth.com/api/classes/AuthenticatedRouteMixin.html). This will make the
+route inaccessible until we finish the authentication. It will automatically redirect you to the specified login route,
+by default `login`, if you are not authenticated.
+
+```js
+// app/routes/index.js
+
+import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+export default Ember.Route.extend(AuthenticatedRouteMixin, {
+});
+```
+
+If you're still running the app when you save this, you will see it redirect to the `login` route.
+
+We'll also designate the `login` route as available for unauthenticated access by applying the
```

Comment:
This should maybe read _"for unauthenticated access __only__"_ as that's really the point of the mixin - making sure authenticated users can __not__ access it.